components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java

```diff
@@ -0,0 +1,157 @@
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * {@code ConfigurationLoader} contains utilities to load configuration file content required for Internal APIs
+ * implementation.
+ */
+public class ConfigurationLoader {
+
+    private static Log log = LogFactory.getLog(ConfigurationLoader.class);
+
+    private static final QName ROOT_Q = new QName("apis");
+    private static final QName HANDLER_Q = new QName("api");
+    private static final QName CLASS_Q = new QName("class");
+    private static final QName NAME_ATT = new QName("name");
+
+    public static List<InternalAPI> loadInternalAPIs(String apiFilePath) {
+
+        List<InternalAPI> internalApis = new ArrayList<>();
+        OMElement apiConfig = MiscellaneousUtil.loadXMLConfig(apiFilePath);
+        if (apiConfig != null) {
+
+            if (!ROOT_Q.equals(apiConfig.getQName())) {
+                handleException("Invalid internal api configuration file");
+            }
+
+            Iterator iterator = apiConfig.getChildrenWithName(HANDLER_Q);
+            while (iterator.hasNext()) {
+                OMElement handlerElem = (OMElement) iterator.next();
+
+                String name = null;
+                if (handlerElem.getAttribute(NAME_ATT) != null) {
+                    name = handlerElem.getAttributeValue(NAME_ATT);
+                } else {
+                    handleException("Name not defined in one or more handlers");
+                }
+
+                if (handlerElem.getAttribute(CLASS_Q) != null) {
+                    String className = handlerElem.getAttributeValue(CLASS_Q);
+                    if (!"".equals(className)) {
+                        InternalAPI internalApi = createAPI(className);
+                        if (internalApi != null) {
+                            internalApis.add(internalApi);
+                            internalApi.setName(name);
+                        }
+                    } else {
+                        handleException("Class name is null for Internal InternalAPI name : " + name);
+                    }
+                } else {
+                    handleException("Class name not defined for Internal InternalAPI named : " + name);
+                }
+            }
+        }
+        return internalApis;
+    }
+
+    private static InternalAPI createAPI(String classFQName) {
+
+        Object obj = null;
+        try {
+            obj = Class.forName(classFQName).newInstance();
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+            handleException("Error creating Internal InternalAPI for class name : " + classFQName, e);
+        }
+
+        if (obj instanceof InternalAPI) {
+            return (InternalAPI) obj;
+        } else {
+            handleException("Error creating Internal InternalAPI. The InternalAPI should be of type " +
+                    "InternalAPI");
+        }
+        return null;
```

Comment:
Should we return an null or would it be better to add exception to method signature?

Reply:
This is to make compiler happy. it will never return a null